app/src/main/java/com/example/traders/watchlist/adapters/ViewPagerAdapter.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.traders.watchlist.adapters
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.example.traders.watchlist.allCrypto.AllCryptoFragment
+import com.example.traders.watchlist.favourites.FavouriteCryptoFragment
+import com.example.traders.watchlist.newCrypto.NewCryptoFragment
+
+class ViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :  FragmentStateAdapter(fragmentManager, lifecycle){
+    override fun getItemCount(): Int {
+        return 3
+    }
+
+    override fun createFragment(position: Int): Fragment {
+        return when(position) {
+            0 -> {
```

Comment:
also, you can create constants for these positions so the code would be more readable.

We create constants in `companion object` - it's a place in a class where all the variables have the same value in all the different instances of the class.
so you could write like this:
```
companion object {
      private const val FIRST_POSITION = 0
      
}
```